lib/types/addons.d.ts

```diff
@@ -0,0 +1,93 @@
+import API, { INormalizeError, IRazorpayQuery } from './api'
+import { IRazorpayItemId } from './items';
+
+
+export interface IRazorpayAddon {
+    /**
+     * A name for the add-on. For example, `Extra appala (papadum)`.
+     */
+    name?: string;
+    /**
+     * The amount you want to charge the customer for the add-on, in the currency subunit. For example, `30000`.
+     */
+    amount?: number;
+    /**
+     * The currency in which the customer should be charged for the add-on. For example, `INR`.
+     */
+    currency?: string;
+    /**
+     * Description for the add-on. For example, `1 extra oil fried appala with meals`
+     */
+    description?: string;
+}
+
+export interface IRazorpayAddonId {
+    /**
+     * The unique identifier of the created add-on.
+     */
+    id: string,
+    /**
+     * The unique identifier of the created add-on.
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏This is same as id?

Reply:
Changed to `Indicates the type of entity.`